needle.pm

```diff
@@ -195,7 +202,9 @@ sub wanted_ {
 }
 
 sub init {
-    $needledir = shift if @_;
+    $needledir    = shift if @_;
+    $shared_cache = shift if @_;
```

Comment:
this is changing the behaviour though. In one cache you leave a previously defined value and in your case it will set it to undef.

I wonder though if we care.